HW23/roles/hw23/tasks/hw23_1.yml

```diff
@@ -0,0 +1,74 @@
+- name: Add Docker on Centos
+  ansible.builtin.get_url:
+    url: https://download.docker.com/linux/centos/docker-ce.repo
+    dest: /etc/yum.repos.d/
+  become: true
+  when: ansible_facts['os_family'] == "RedHat"
+
+- name: Add Docker on Ubuntu
+  block:
+    - name: Add Docker GPG key
+      ansible.builtin.apt_key:
+        url: https://download.docker.com/linux/ubuntu/gpg
+        state: present
+    - name: Add Docker repo
+      ansible.builtin.apt_repository:
+        repo: deb https://download.docker.com/linux/ubuntu focal stable
+        state: present
+  become: true
+  when: ansible_facts['os_family'] == "Debian" 
+
+- name: Install packages
+  ansible.builtin.package:
+    name: "{{  item  }}"
+    state: present
+  become: true
+  loop: "{{  packages  }}"
+
+- name: Start and enable the Docker daemon and nginx
+  ansible.builtin.service:
+    name: "{{  item  }}"
+    state: started
+    enabled: yes
+  become: true
+  loop:
+    - docker
+    - nginx
+
+- name: Var rc=0
```

Comment:
For task 3b please use **register**  and **when**. For that you can create two bash command to generate code 0 and 2.

Reply:
Command: The command(s) will not be processed through the shell, so variables like $HOSTNAME and operations like "*", "<", ">", "|", ";" and "&" will not work. Use the ansible.builtin.shell module if you need these features.
Shell: It is almost exactly like the ansible.builtin.command module but runs the command through a shell (/bin/sh) on the remote node.
Raw: Executes a low-down and dirty SSH command, not going through the module subsystem. This is useful and should only be done in a few cases. A common case is installing python on a system without python installed by default. Another is speaking to any devices such as routers that do not have any Python installed. In any other case, using the ansible.builtin.shell or ansible.builtin.command module is much more appropriate.

Reply:
Done
![image](https://user-images.githubusercontent.com/44346029/122265872-d9aeb100-cee1-11eb-9302-4cbe45dfed02.png)